src/main/kotlin/org/dif/crypto/key/RecipientKeySelector.kt

```diff
@@ -21,4 +24,55 @@ class RecipientKeySelector(private val didDocResolver: DIDDocResolver, private v
             didDoc.findVerificationMethod(verificationMethodId)
         }
     )
```

Comment:
In `verifyKey`: no need to call `didDocResolver.resolve` if `isDIDFragment(selector)` is true.